src/arch/cuda/hybridAPI/hapi_impl.cpp

```diff
@@ -89,15 +93,89 @@ CsvDeclare(GPUManager, gpu_manager);
 CpvDeclare(int, my_device); // GPU device that this thread is mapped to
 CpvDeclare(bool, device_rep); // Is this PE a device representative thread? (1 per device)
 
+// Returns the local rank of the logical node (process) that the given PE belongs to
```

Comment:
Should we move these to machine-common-core.C? That's where the other similar functions live.

Reply:
I think these can live here for now since HAPI is the only one using them.